cloudinit/config/cc_ca_certs.py

```diff
@@ -162,20 +169,33 @@ def handle(name, cfg, cloud, log, _args):
     @param log: Pre-initialized Python logger object to use for logging.
     @param args: Any module arguments from cloud.cfg
     """
-    # If there isn't a ca-certs section in the configuration don't do anything
-    if "ca-certs" not in cfg:
+    if "ca-certs" in cfg:
+        log.warning(
+            "DEPRECATION: key 'ca-certs' is now deprecated. Use 'ca_certs' by"
+            " version 23.1."
+        )
+        ca_cert_cfg = cfg["ca-certs"]
+    elif "ca_certs" not in cfg:
         log.debug(
             "Skipping module named %s, no 'ca-certs' key in configuration",
             name,
         )
         return
-
-    ca_cert_cfg = cfg["ca-certs"]
+    else:
+        ca_cert_cfg = cfg["ca_certs"]
     distro_cfg = _distro_ca_certs_configs(cloud.distro.name)
 
-    # If there is a remove-defaults option set to true, remove the system
+    # If there is a remove_defaults option set to true, remove the system
     # default trusted CA certs first.
-    if ca_cert_cfg.get("remove-defaults", False):
+    if "remove-defaults" in ca_cert_cfg:
+        log.warning(
+            "DEPRECATION: key 'ca-certs.remove-defaults' is now deprecated."
+            " Use 'ca_certs.remove_defaults' by version 23.1."
```

Comment:
Same thing about the date here

Reply:
+1. as it is currently. I've just added the DEPRECATION warning and dropped the version suggestion from this PR. We can sort how/when to deprecate a number of schema related values as a separate PR since there will be multiple cases by the time we get through all cc_* modules.